crates/anvil/core/src/eth/transaction/mod.rs

```diff
@@ -682,8 +682,36 @@ impl TypedTransaction {
     }
 
     /// Max cost of the transaction
+    /// It is the gas limit multiplied by the gas price,
+    /// and if the transaction is EIP-4844, the result of (total blob gas cost * max fee per blob
+    /// gas) is also added
     pub fn max_cost(&self) -> U256 {
-        self.gas_limit().saturating_mul(self.gas_price())
+        let mut max_cost = self.gas_limit().saturating_mul(self.gas_price());
+
+        if self.is_eip4844() {
+            max_cost = max_cost.saturating_add(
+                self.blob_gas()
+                    .map(U256::from)
+                    .unwrap_or(U256::ZERO)
+                    .mul(self.max_fee_per_blob_gas().unwrap_or(U256::ZERO)),
+            )
+        }
+
+        max_cost
+    }
+
+    pub fn blob_gas(&self) -> Option<u64> {
+        match self {
```

Comment:
tx.tx().tx() lol

Reply:
not great but maybe we need to life with it cc @mattsse  - this was already here